contracts/src/v0.8/l2ep/dev/scroll/ScrollSequencerUptimeFeed.sol

```diff
@@ -0,0 +1,242 @@
+    s_feedState.updatedAt = updatedAt;
+    emit RoundUpdated(_getStatusAnswer(status), updatedAt);
+  }
+
+  /// @notice Record a new status and timestamp if it has changed since the last round.
+  /// @dev This function will revert if not called from `l1Sender` via the L1->L2 messenger.
+  ///
+  /// @param status Sequencer status
+  /// @param timestamp Block timestamp of status update
+  function updateStatus(bool status, uint64 timestamp) external override {
+    FeedState memory feedState = s_feedState;
+    if (
+      msg.sender != address(s_l2CrossDomainMessenger) || s_l2CrossDomainMessenger.xDomainMessageSender() != s_l1Sender
+    ) {
+      revert InvalidSender();
+    }
+
+    // Ignore if latest recorded timestamp is newer
+    if (feedState.startedAt > timestamp) {
+      emit UpdateIgnored(feedState.latestStatus, feedState.startedAt, status, timestamp);
+      return;
+    }
+
+    if (feedState.latestStatus == status) {
+      _updateRound(feedState.latestRoundId, status);
+    } else {
+      feedState.latestRoundId += 1;
+      _recordRound(feedState.latestRoundId, status, timestamp);
+    }
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestAnswer() external view override checkAccess returns (int256) {
+    FeedState memory feedState = s_feedState;
+    return _getStatusAnswer(feedState.latestStatus);
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestTimestamp() external view override checkAccess returns (uint256) {
+    FeedState memory feedState = s_feedState;
+    return feedState.startedAt;
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestRound() external view override checkAccess returns (uint256) {
+    FeedState memory feedState = s_feedState;
+    return feedState.latestRoundId;
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function getAnswer(uint256 roundId) external view override checkAccess returns (int256) {
+    if (_isValidRound(roundId)) {
+      return _getStatusAnswer(s_rounds[uint80(roundId)].status);
+    }
+
+    revert NoDataPresent();
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function getTimestamp(uint256 roundId) external view override checkAccess returns (uint256) {
+    if (_isValidRound(roundId)) {
+      return s_rounds[uint80(roundId)].startedAt;
+    }
+
+    revert NoDataPresent();
+  }
+
+  /// @inheritdoc AggregatorV3Interface
+  function getRoundData(
+    uint80 _roundId
+  )
+    public
+    view
+    override
+    checkAccess
+    returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+  {
+    if (!_isValidRound(_roundId)) {
+      revert NoDataPresent();
+    }
+
+    Round memory round = s_rounds[_roundId];
+    answer = _getStatusAnswer(round.status);
+    startedAt = uint256(round.startedAt);
+    roundId = _roundId;
+    updatedAt = uint256(round.updatedAt);
+    answeredInRound = roundId;
+
+    return (roundId, answer, startedAt, updatedAt, answeredInRound);
```

Comment:
Let's also just return the values here, instead of assigning them and then returning